src/pybind/mgr/cephadm/module.py

```diff
@@ -1810,11 +1824,16 @@ def _check_daemons(self):
                 continue
 
             # dependencies?
-            if dd.daemon_type == 'grafana':
-                # put running instances at the front of the list
-                grafanas.insert(0, dd)
-            elif dd.daemon_type == 'iscsi':
-                iscsi_daemons.append(dd)
+            if dd.daemon_type in ['grafana', 'iscsi', 'prometheus', 'alertmanager']:
+                cephadm_service = self.cephadm_services.get(dd.daemon_type)
+                v = post_actions.setdefault(dd.daemon_type,
```

Comment:
If we want a dictionary with a default value for undefined keys, why not using `DefaultDict` instead?

Reply:
Thanks, fixed.